### ISSUE TITLE
`ef migrations script --idempotent` generates broken script

### ISSUE THREAD

**Issue by Sa1Gur** [dotnet/efcore#27645]
After updating Microsoft.EntityFrameworkCore.Design and Npgsql.EntityFrameworkCore.PostgreSQL to the version 6.0.* the script generated by  `dotnet ef migrations --project  script --idempotent` became broken.

In 5.0.* it was 
```
....
DO $$
BEGIN
    IF NOT EXISTS(SELECT 1 FROM "__EFMigrationsHistory" WHERE "MigrationId" = '20210806071539_InitialCreate') THEN
    CREATE SCHEMA IF NOT EXISTS dbo;
    END IF;
END $$;
``` 

In 6.0.*

```
...
DO $EF$
BEGIN
    IF NOT EXISTS(SELECT 1 FROM "__EFMigrationsHistory" WHERE "MigrationId" = '20210806071539_InitialCreate') THEN
    DO $EF$
    BEGIN
        IF NOT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname = 'dbo') THEN
            CREATE SCHEMA dbo;
        END IF;
    END $EF$;
    END IF;
END $EF$;
```

The second `DO $EF$` results error. Two scripts attached and the project to reproduce issues https://github.com/Sa1Gur/Repository.
just run 
```
dotnet ef migrations --project .\mdp2repository_sql\mdp2repository_sql.csproj script --idempotent --output "version5.sql"
[dotnet](url) ef migrations --project .\mdp2repository_db\mdp2repository_db.csproj script --idempotent --output "version6.sql"
```

Database provider: Npgsql.EntityFrameworkCore.PostgreSQL
Target framework:  NET 6.0
Operating system: Windows 11
[version5.txt](https://github.com/dotnet/efcore/files/8251796/version5.txt)
[version6.txt](https://github.com/dotnet/efcore/files/8251797/version6.txt)

IDE: dotnet cli

**Comment by roji** [comment]
This has already been fixed for the upcoming 6.0.4 release.

Note that this is a PostgreSQL-specific issue, and so belongs in https://github.com/npgsql/efcore.pg rather than here.

**Comment by Sa1Gur** [comment]
Sorry about that (I"ve got some doubts where to post and decided wrong). Thank you so much for reply and great work!
